docs/typescript.md

```diff
@@ -1,14 +1,209 @@
-# TypeScript Guidelines
+# Typescript Guidelines
 
-These guidelines specifically apply to TypeScript.
+**_Purpose_**: Establish best practices for writing robust, maintainable, and efficient TypeScript code.
 
-## Provide explicit return types for functions/methods
+**_Audience_**: Internal and external developers using TypeScript in Metamask.
```

Comment:
```suggestion
**_Audience_**: Internal and external developers using TypeScript to contribute to MetaMask.
```

We want to use consistent styling for our main product name (similarly, "Consensys," not "ConsenSys" or some other variation).